include/openthread/platform/radio.h

```diff
@@ -621,6 +622,15 @@ bool otPlatRadioGetPromiscuous(otInstance *aInstance);
  */
 void otPlatRadioSetPromiscuous(otInstance *aInstance, bool aEnable);
 
+/**
+ * Indicates whether radio should stay in Receive or Sleep state during idle periods.
```

Comment:
What is the definition of "idle period" from prespective of `otPlatRadio` API?

We have this state machine for radio states and APIs to put the radio in different states.
```
/**
 * The following are valid radio state transitions:
 *
 *                                    (Radio ON)
 *  +----------+  Enable()  +-------+  Receive() +---------+   Transmit()  +----------+
 *  |          |----------->|       |----------->|         |-------------->|          |
 *  | Disabled |            | Sleep |            | Receive |               | Transmit |
 *  |          |<-----------|       |<-----------|         |<--------------|          |
 *  +----------+  Disable() +-------+   Sleep()  +---------+   Receive()   +----------+
 *                                    (Radio OFF)                 or
 *                                                        signal TransmitDone
 *
 * During the IEEE 802.15.4 data request command the transition Sleep->Receive->Transmit
 * can be shortened to direct transition from Sleep to Transmit if the platform supports
 * the OT_RADIO_CAPS_SLEEP_TO_TX capability.
 */
```

Is the "idle period" refer to when radio is "Tx" state and perofmring CMSA backoffs?

Reply:
I'm not sure if the diagram is already outdated even before this PR, for instance when comes to delayed rx/tx.

Regarding CMSA backoffs, I don't have a strong opinion, but it seemed a good idea to me to also consider idle period this case, which in turn simplifies the code. At the end, Mac translates `mRxOnWhenIdle` to `mRxOnWhenBackoff` directly.

Reply:
I think this is still valid and used.The delayed tx/rx (for CSL) is  happening in parallel to this state machine. From our perspective radio is one of these states, but we can use API to schedule things to happen at certain time which should happen over this state machine.